libraries/aws.rb

```diff
@@ -93,7 +93,10 @@ def specific_machine_options(component, _count = nil)
         return [] unless @node['delivery-cluster'][component]
         options = []
         options << { bootstrap_options: { instance_type: @node['delivery-cluster'][component]['flavor'] } } if @node['delivery-cluster'][component]['flavor']
+        options << { bootstrap_options: { security_group_ids: @node['delivery-cluster'][component]['security_group_ids'] } } if @node['delivery-cluster'][component]['security_group_ids']
+        options << { image_id: @node['delivery-cluster'][component]['image_id'] } if @node['delivery-cluster'][component]['image_id']
         # Specify more specific machine_options to add
+        options
```

Comment:
I'm not nearly a strong enough rubyist to tell you why, but in testing I found that if the last element in the options is undefined, the function gets angry at me and throws nilClass errors. Having the final options line for the return seems to have fixed that -- I was able to get it working with any number of the properties defined in attributes, and failed back to the aws attributes otherwise.